app/services/cycle_timetable.rb

```diff
@@ -95,6 +95,10 @@ class CycleTimetable
   }.freeze
 
   def self.current_year(now = Time.zone.now)
+    if ActiveRecord::Base.connected? && current_cycle_schedule.in?(%i[today_is_after_find_opens today_is_after_apply_opens])
```

Comment:
`current_year` gets called in an initializer, `business_time.rb` I think. I looked at wrapping the initializer in an on_load hook but didn't spend enough time on it because I was focusing on getting everything working.

But this is maybe something that should be looked at immediately after this is merged.

Along with the performance of the application. As I said in the description, `current_year` gets called a lot and now it makes a DB request.